lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -229,7 +230,7 @@ private function getLastModification(ReflectionClass $class): int
         $parent = $class->getParentClass();
 
         $lastModification =  max(array_merge(
-            [$filename ? filemtime($filename) : 0],
+            [is_file($filename) ? filemtime($filename) : 0],
```

Comment:
`$filename` is whatever [`ReflectionClass::getFileName()`](https://www.php.net/manual/de/reflectionclass.getfilename.php) returns and that could technically be `false`. I don't want to pass `false` to `is_file()`.

```suggestion
            [$filename !== false && is_file($filename) ? filemtime($filename) : 0],
```

That being said, what actually does `$filename` contain in case of a class that comes from eval'ed code? 🤔

Reply:
1) Thank you for fast answer and support.
2) I added check `$filename !== false`.
3) > That being said, what actually does $filename contain in case of a class that comes from eval'ed code?

The `filename` contains something like (see Issue [#432](https://github.com/doctrine/annotations/issues/492)): `/home/.../my-project/vendor/lexik/jwt-authentication-bundle/Security/Authenticator/ForwardCompatAuthenticatorTrait.php(14)`.